packages/x-date-pickers/src/locales/faIR.ts

```diff
@@ -45,18 +51,18 @@ const faIRPickers: Partial<PickersLocaleText<any>> = {
   // Clock labels
   clockLabelText: (view, time, adapter) =>
     ` را انتخاب کنید ${timeViews[view]}. ${time === null ? 'هیچ ساعتی انتخاب نشده است' : `ساعت انتخاب ${adapter.format(time, 'fullTime')} می باشد`}`,
-  hoursClockNumberText: (hours) => `${hours} ساعت‌ها`,
-  minutesClockNumberText: (minutes) => `${minutes} دقیقه‌ها`,
-  secondsClockNumberText: (seconds) => `${seconds} ثانیه‌ها`,
+  hoursClockNumberText: (hours) => `${latinToPersianDigits(hours)} ساعت‌ها`,
+  minutesClockNumberText: (minutes) => `${latinToPersianDigits(minutes)} دقیقه‌ها`,
+  secondsClockNumberText: (seconds) => `${latinToPersianDigits(seconds)} ثانیه‌ها`,
 
   // Digital clock labels
   selectViewText: (view) => ` را انتخاب کنید ${timeViews[view]}`,
 
   // Calendar labels
   calendarWeekNumberHeaderLabel: 'عدد هفته',
-  calendarWeekNumberHeaderText: '#',
-  calendarWeekNumberAriaLabelText: (weekNumber) => `هفته ${weekNumber}`,
-  calendarWeekNumberText: (weekNumber) => `${weekNumber}`,
+  calendarWeekNumberHeaderText: latinToPersianDigits('#'),
```

Comment:
Is this correct? Doesn't seem like `#` will get replaced by a number

Reply:
But even the other changes, it looks strange. It's related to https://github.com/mui/mui-x/issues/13222. We could start by using .toLocaleString('fa-IR'), it's not super fast, but might be fine, and later, say, with Base UI having a locale number formater, rely on it.